### ISSUE TITLE
java.util.NoSuchElementException: None.get using json macro

### ISSUE THREAD

**Issue by srenault** [propensive/rapture#198]
I think there is a regression somewhere. I'm unable to do this kind of thing:

```
scala> import rapture.json._, jsonBackends.jawn._
scala> val body = json"""{ "statements": ${json"{}"} }"""
```

```
java.util.NoSuchElementException: None.get
  at scala.None$.get(Option.scala:347)
  at scala.None$.get(Option.scala:345)
  ... 43 elided
```

**Comment by srenault** [comment]
The issue is we try to parse `{ "statements": json"""{}""" }`.
This seems to be appear since this commit: https://github.com/propensive/rapture/commit/3e67e3d67048e1595f9e78606a3876f213e0d499
But I don't think this is where we have to fix this.
I've the feeling that the real problem is the use of `toString` here:
https://github.com/propensive/rapture/blob/dev/data/shared/src/main/scala/rapture/data/context.scala#L103

What do you think?

**Comment by propensive** [comment]
Thanks for the pointers on this, @srenault - that will make it a lot easier for me to find the bug! I had noticed the problem occurring in the REPL, but also noticed that the stack track mentioned no Rapture code, so I was confused...

But I'll look at it now.

**Comment by srenault** [comment]
One other issue I noticed  when I use the `json` context. It's about Double and precision.

``` scala
import rapture.json._, jsonBackends.jawn._
import rapture.json.formatters.compact._
import rapture.core.decimalFormats.exact._

val pi = json"""3.14159265359"""
Json.format(pi)
// res8: String = 3.14159265359 <= OK

val x = json"""{"pi": $pi}"""
Json.format(x)
// res9: String = {"pi":3.142} <= NOT OK
```

**Comment by propensive** [comment]
I've just fixed this issue. I'll create a new one for the precision problem...

**Comment by hrj** [comment]
Is this fix not included in the latest release (v2.0.0-M6)?

If not, please release a new version as soon as possible.
